internal/controller/binding/binding.go

```diff
@@ -0,0 +1,359 @@
+	cr, ok := mg.(*v1alpha1.Binding)
+	if !ok {
+		return managed.ExternalDelete{}, errors.New(errNotBinding)
+	}
+	fmt.Printf("Deleting binding: %+v\n", cr.Name)
+
+	bindingInfo := GenerateBindingInfo(&cr.Spec.ForProvider)
+
+	resp, err := c.service.Rmqc.DeleteBinding(cr.Spec.ForProvider.Vhost, bindingInfo)
+
+	if err != nil {
+		fmt.Printf("Error deleting binding: %+v\n", err)
+		return managed.ExternalDelete{}, errors.Wrap(err, errDeleteFailed)
+	}
+	if err := resp.Body.Close(); err != nil {
+		fmt.Printf("Error closing response body: %v\n", err)
+	}
+
+	return managed.ExternalDelete{}, nil
+}
+func (c *external) Disconnect(ctx context.Context) error {
+	return nil
+}
+
+func GenerateBindingInfo(binding *v1alpha1.BindingParameters) rabbithole.BindingInfo {
+	bidingInfo := rabbithole.BindingInfo{
+		Source:          binding.Source,
+		Destination:     binding.Destination,
+		DestinationType: binding.DestinationType,
+		RoutingKey:      binding.RoutingKey,
+		// RabbitMQ API: /api/bindings/{vhost}/e/{source}/{destination_type}/{destination}/routing_key
+		// Library Delete method: /api/bindings/{vhost}/e/{source}/{destination_type}/{destination}/{properties_keys}
+		// Then: PropertiesKey = RoutingKey
+		PropertiesKey: binding.RoutingKey,
+	}
+	return bidingInfo
+}
+
+func GenerateBindingObservation(api *rabbithole.BindingInfo) v1alpha1.BindingObservation {
+	if api == nil {
+		return v1alpha1.BindingObservation{}
+	}
+	binding := v1alpha1.BindingObservation{
+		Source:          api.Source,
+		Vhost:           api.Vhost,
+		Destination:     api.Destination,
+		DestinationType: api.DestinationType,
+		RoutingKey:      api.RoutingKey,
+	}
+
+	return binding
+}
+
+func isUpToDate(spec *v1alpha1.BindingParameters, api *rabbithole.BindingInfo) bool { //nolint:gocyclo
+
+	if spec.Source != api.Source {
+		return false
+	}
+	if spec.Vhost != api.Vhost {
+		return false
+	}
+	if spec.Destination != api.Destination {
+		return false
+	}
+	if spec.DestinationType != api.DestinationType {
+		return false
+	}
+	if spec.RoutingKey != api.RoutingKey {
+		return false
+	}
+	return true
+}
+
+func listBindings(spec *v1alpha1.BindingParameters, api *rabbitmqclient.RabbitMqService) (bindings []rabbithole.BindingInfo, err error) {
+	switch spec.DestinationType {
+	case "queue":
+		bindings, err = api.Rmqc.ListQueueBindingsBetween(spec.Vhost, spec.Source, spec.Destination)
+	case "exchange":
+		bindings, err = api.Rmqc.ListExchangeBindingsBetween(spec.Vhost, spec.Source, spec.Destination)
+	default:
+		bindings, err = api.Rmqc.ListBindingsIn(spec.Vhost)
+
+	}
+	return bindings, err
+}
+
+func getBinding(bindings []rabbithole.BindingInfo, cr *v1alpha1.Binding) *rabbithole.BindingInfo {
+	for _, binding := range bindings {
+		if binding.Source == cr.Spec.ForProvider.Source && binding.Destination == cr.Spec.ForProvider.Destination && binding.DestinationType == cr.Spec.ForProvider.DestinationType {
```

Comment:
So the only think to potential update will be routing-key - is that correct (as other parameters are kind-of complex id key)